test/ClangImporter/indirect_field_codegen.swift

```diff
@@ -0,0 +1,66 @@
+// RUN: %target-swift-frontend -emit-silgen -I %S/Inputs/custom-modules %s | %FileCheck %s
+
+// https://bugs.swift.org/browse/SR-10571
+
+import IndirectFields
+
+extension StructWithIndirectField2Copy {
+  init(x: UInt32, y: UInt32) {
+    self.init(.init(v: x + y))
+  }
+
+  init(s: StructWithIndirectField2) {
+    self.init(x: s.x, y: s.y)
+  }
+}
+
+// StructWithIndirectField2Copy.__Unnamed_union___Anonymous_field0.v.setter
+// CHECK-DAG: sil shared [transparent] [serializable] [ossa] @$sSo28StructWithIndirectField2CopyV34__Unnamed_union___Anonymous_field0V1vs6UInt32Vvs : $@convention(method) (UInt32, @inout StructWithIndirectField2Copy.__Unnamed_union___Anonymous_field0)
+
+// StructWithIndirectField2.x.getter
+// CHECK-DAG: sil shared [transparent] [serializable] [ossa] @$sSo24StructWithIndirectField2V1xs6UInt32Vvg : $@convention(method) (StructWithIndirectField2) -> UInt32
+
+// StructWithIndirectField2.__Unnamed_union___Anonymous_field0.x.getter
+// CHECK-DAG: sil shared [transparent] [serializable] [ossa] @$sSo24StructWithIndirectField2V34__Unnamed_union___Anonymous_field0V1xs6UInt32Vvg : $@convention(method) (StructWithIndirectField2.__Unnamed_union___Anonymous_field0) -> UInt32
+
+// StructWithIndirectField2.__Unnamed_union___Anonymous_field0.__Anonymous_field1.getter
+// CHECK-DAG: sil shared [transparent] [serializable] [ossa] @$sSo24StructWithIndirectField2V34__Unnamed_union___Anonymous_field0V02__G7_field1AD02__e10_struct___g1_I0Vvg : $@convention(method) (StructWithIndirectField2.__Unnamed_union___Anonymous_field0) -> StructWithIndirectField2.__Unnamed_union___Anonymous_field0.__Unnamed_struct___Anonymous_field1
+
+// $So24StructWithIndirectField2V34__Unnamed_union___Anonymous_field0V02__e10_struct___G7_field1V$x$getter
+// sil shared @$So24StructWithIndirectField2V34__Unnamed_union___Anonymous_field0V02__e10_struct___G7_field1V$x$getter : $@convention(c) (StructWithIndirectField2.__Unnamed_union___Anonymous_field0.__Unnamed_struct___Anonymous_field1) -> UInt32
```

Comment:
Missing `CHECK-DAG` for this one. Also the comment could probably be improved to clarify why it doesn't demangle nicely.

Reply:
1. There is a setter on lines 17-18. Did you mean some other kind of setter? 
2. I will fix the missing CHECK-DAG bit. 
3. For the comments, I've merely copy-pasted the output from `-emit-silgen`, since it also emits comments. To be honest, I'm surprised at this behavior too :sweat_smile:, not sure why that's happening.

Reply:
That's the setter for the property VarDecl, but it isn't one of these synthesized setters. (You can tell by the name of the function not having `$setter` on the end.)

`-emit-silgen` will demangle names that look like Swift mangled names, but this function *doesn't* look like a Swift mangled name, even though it happens to have most of one in it. (It doesn't have the right prefix and it has extra stuff on the end.)

Reply:
1. "doesn't have the right prefix" == "doesn't begin with a "$s" (I skimmed [Mangling.rst](https://github.com/apple/swift/blob/master/docs/ABI/Mangling.rst))?
2. "extra stuff" = "$x$getter and similar"?
3. Should I get rid of the checks for the property getters/setters? Seems unnecessarily confusing -- we don't really intend to test them, as my change doesn't affect their use.
4. I am not quite sure how to phrase an explanation. How about "The Swift demangler doesn't understand the naming scheme for accessor functions, so the names aren't demangled nicely."? Maybe I should just delete the comments with the mangled names as they don't add anything?

Reply:
Yep to 1 and 2. The `$` that's there is almost coincidental; it's just supposed to be an unlikely separator to not collide with any possible user-defined names. (The mangling prefix didn't used to have a `$` in it, after all.)

3 seems reasonable to me. For 4, I think the comments with the mangled names don't help, but if there's another comment that *would* help you can include that. Whatever you think will make the test case easier to understand by someone reading it in the future!

Reply:
Fixed.